lib/goo/sparql/mixins/solution_lang_filter.rb

```diff
@@ -15,26 +15,36 @@ def main_lang_filter(id, attr, value)
           [index, value]
         end
 
-        def fill_models_with_other_languages(models_by_id, list_attributes)
-          @other_languages_values.each  do |id, languages_values|
-            languages_values.each do |attr, index_values|
-              model_attribute_val = models_by_id[id].instance_variable_get("@#{attr.to_s}")
-              values = languages_values_to_set(index_values, model_attribute_val)
-              m = models_by_id[id]
-              value = nil
-              is_struct = m.respond_to?(:klass)
-              if !values.nil? && list_attributes.include?(attr)
-                value = values || []
-
-              elsif !values.nil?
-                value = values.first || nil
-              end
+        def find_model_objects_by_lang(objects_by_lang, lang, model_id, predicate)
+          objects_by_lang[lang]&.find { |obj| obj[:id].eql?(model_id) && obj[:predicate].eql?(predicate) }
+        end
+
+        def fill_models_with_other_languages(models_by_id, objects_by_lang, list_attributes, attributes,klass)
+      
+          other_platform_languages = [:EN, :FR]
```

Comment:
Why here other_platform_languages equal to `:EN` and `:FR` ? This need to be generic a portal can have other languages for example german or other

Reply:
yes , i hard coded the value here